address-model-lib/src/main/java/io/enmasse/address/model/KubeUtil.java

```diff
@@ -8,22 +8,33 @@
  * Varioius static utilities that don't belong in a specific place
  */
 public class KubeUtil {
+    private static int MAX_KUBE_NAME = 63 - 3; // max length of identifier - space for pod identifier
     public static String sanitizeName(String name) {
         String clean = name.toLowerCase().replaceAll("[^a-z0-9\\-]", "");
+        System.out.println(clean);
         if (clean.startsWith("-")) {
             clean = clean.replaceFirst("-", "1");
         }
 
-        if (clean.length() > 60) {
-            clean = clean.substring(0, 60);
+        if (clean.length() > MAX_KUBE_NAME) {
+            clean = clean.substring(0, MAX_KUBE_NAME);
         }
 
         if (clean.endsWith("-")) {
-            clean = clean.substring(0, clean.length() - 2) + "1";
+            clean = clean.substring(0, clean.length() - 1) + "1";
         }
         return clean;
     }
 
+    public static String sanitizeWithUuid(String name, String uuid) {
+        name = sanitizeName(name);
+        if (name.length() + uuid.length() > MAX_KUBE_NAME) {
```

Comment:
Is this missing '+1' to account for the separator '-'? (the chars reserved for pod id will mean it doesn't exceed an invalid name but it would restrict to 9 pods - probably still plenty for most cases of course).

Reply:
@grs Good catch, thanks!